sycl/test-e2e/USM/badmalloc.cpp

```diff
@@ -36,10 +36,12 @@ int main(int argc, char *argv[]) {
   std::cout << "p = " << p << std::endl;
   if (p != nullptr)
     return 3;
-  p = malloc(-1, q, usm::alloc::shared);
-  std::cout << "p = " << p << std::endl;
-  if (p != nullptr)
-    return 4;
+  if (q.get_device().has(aspect::usm_shared_allocations)) {
```

Comment:
This all looks good, but shouldn't we also be testing the case of calling malloc when we _don't_ have the aspect and then checking that the correct exception has been thrown?  I know this would be a speculative test, as I'm not sure that we yet have a device that doesn't have the aspect, but we should be testing that error affirmatively.

Reply:
Good catch, I'll add some logic here to test that case!